code/datums/storage/subtypes/bag.dm

```diff
@@ -0,0 +1,173 @@
+/datum/storage/bag
+	allow_quick_gather = 1
+	allow_quick_empty = 1
+	display_contents_with_number = 0 // UNStABLE AS FuCK, turn on when it stops crashing clients
+	use_to_pickup = 1
+
+/datum/storage/bag/trash
+	max_w_class = WEIGHT_CLASS_SMALL
+	storage_slots = 21
+
+/datum/storage/bag/trash/New(atom/parent)
+	. = ..()
+	set_holdable(cant_hold_list = list(/obj/item/disk/nuclear))
+
+/datum/storage/bag/plasticbag
+	max_w_class = WEIGHT_CLASS_SMALL
+	storage_slots = 21
+
+/datum/storage/bag/plasticbag/New(atom/parent)
+	. = ..()
+	set_holdable(cant_hold_list = list(/obj/item/disk/nuclear))
+
+/datum/storage/bag/ore
+	storage_slots = 50
+	max_storage_space = 200 //Doesn't matter what this is, so long as it's more or equal to storage_slots * ore.w_class
+	max_w_class = WEIGHT_CLASS_NORMAL
+
+/datum/storage/bag/ore/New(atom/parent)
+	. = ..()
+	set_holdable(list(/obj/item/ore))
+
+/datum/storage/bag/plants
+	storage_slots = 50; //the number of plant pieces it can carry.
+	max_storage_space = 200 //Doesn't matter what this is, so long as it's more or equal to storage_slots * plants.w_class
+	max_w_class = WEIGHT_CLASS_NORMAL
+
+/datum/storage/bag/plants/New(atom/parent)
+	. = ..()
+	set_holdable(list(
+		/obj/item/reagent_containers/food/snacks/grown,
+		/obj/item/seeds,
+		/obj/item/grown,
+	))
+
+/datum/storage/bag/cash
+	storage_slots = 50; //the number of cash pieces it can carry.
+	max_storage_space = 200 //Doesn't matter what this is, so long as it's more or equal to storage_slots * cash.w_class
+	max_w_class = WEIGHT_CLASS_NORMAL
+
+/datum/storage/bag/cash/New(atom/parent)
+	. = ..()
+	set_holdable(list(
+		/obj/item/coin,
+		/obj/item/spacecash,
+	))
+
+/datum/storage/bag/sheetsnatcher
+	allow_quick_empty = 1 // this function is superceded
+
+/datum/storage/bag/sheetsnatcher/can_be_inserted(obj/item/W as obj, stop_messages = 0)
+	var/obj/item/storage/bag/sheetsnatcher/sheetsnatcher = parent
+	if(!istype(W,/obj/item/stack/sheet) || istype(W,/obj/item/stack/sheet/mineral/sandstone) || istype(W,/obj/item/stack/sheet/wood))
+		if(!stop_messages)
+			to_chat(usr, "The snatcher does not accept [W].")
+		return 0 //I don't care, but the existing code rejects them for not being "sheets" *shrug* -Sayu
+	var/current = 0
+	for(var/obj/item/stack/sheet/S in parent.contents)
+		current += S.amount
+	if(sheetsnatcher.capacity == current)//If it's full, you're done
+		if(!stop_messages)
+			to_chat(usr, span_warning("The snatcher is full."))
+		return 0
+	return 1
+
+
+// Modified handle_item_insertion.  Would prefer not to, but...
+/datum/storage/bag/sheetsnatcher/handle_item_insertion(obj/item/W, prevent_warning = 0, mob/user)
+	var/obj/item/storage/bag/sheetsnatcher/sheetsnatcher = parent
+	var/obj/item/stack/sheet/S = W
+	if(!istype(S)) return 0
+
```

Comment:
improperly formattted indentation

rest of the proc also has bools showing as ints, including return values